src/WebJobs.Extensions.DurableTask/Auth/ITokenCredentialFactory.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Threading;
+using Microsoft.Extensions.Configuration;
+using Microsoft.WindowsAzure.Storage.Auth;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask.Auth
+{
+    /// <summary>
+    /// Defines methods for retrieving <see cref="TokenCredential"/>.
+    /// </summary>
+    internal interface ITokenCredentialFactory
+    {
+        /// <summary>
+        /// Creates a <see cref="TokenCredential"/> based on the specified <paramref name="configuration"/>.
+        /// </summary>
+        /// <param name="configuration">A configuration containing information about the connection.</param>
+        /// <param name="cancellationToken">
+        /// An optional <see cref="CancellationToken"/> for prematurely canceling the operation.
+        /// </param>
+        /// <returns>
+        /// The <see cref="TokenCredential"/> based on the specified <paramref name="configuration"/>.
+        /// </returns>
+        /// <exception cref="OperationCanceledException">The operation was canceled.</exception>
+        TokenCredential Create(IConfiguration configuration, CancellationToken cancellationToken = default);
```

Comment:
I created two for the purpose of providing defaults. Alternatively, we could make the parameter types `TimeSpan?` and use a default if they value is `null`, but I thought instead to express the optional values through 2 methods. I based these default values from what I saw in the `Azure.Identity` library for a similar class, but since we're not even using one of the overloads, maybe it's overkill to provide it haha